corehq/ex-submodules/pillowtop/processors/form.py

```diff
@@ -0,0 +1,33 @@
+from .interface import PillowProcessor
+from corehq.apps.app_manager.dbaccessors import get_app
+from corehq.util.quickcache import quickcache
+
+
+class AppFormSubmissionTrackerProcessor(PillowProcessor):
+    """
+    Processor used to process each form and mark the corresponding application as
+    having submissions (has_submissions = True).
+    """
+
+    def process_change(self, pillow_instance, change):
+        if change.deleted:
+            return
+
+        doc = change.get_document()
+        if not doc:
+            return
+
+        build_id = doc.get('build_id')
+        domain = change.metadata.domain
+
+        if build_id and domain:
+            app = self._get_app(domain, build_id)
+
+            if not app.has_submissions:
+                app.has_submissions = True
+                app.save()
+                self._get_app.clear(self, domain, build_id)
+
+    @quickcache(['domain', 'build_id'], timeout=60 * 60)
+    def _get_app(self, domain, build_id):
```

Comment:
https://github.com/dimagi/commcare-hq/pull/11819/commits/dbe1b914d98fc80cf6d6cf85831502e1ea93ccb0
is this what you were thinking? this way we only grab the app if it hasn't already been marked